File: generate_en_faqs.js
import { execSync } from 'child_process';
import * as fs from 'fs';

const roFaqPath = 'C:/Users/User/plizio-repo/lib/visualLab/data/faq/roFaq.ts';
const content = fs.readFileSync(roFaqPath, 'utf8');

// Use a regex to match the POI entries
const poiRegex = /"([^"]+)": \{([\s\S]*?)\n  \},/g;
let newContent = content;

const matchPromises = [];

// Since we have all POIs, we need to generate English translation for each RO FAQ
// I will create a script to generate English translations using a simple mapping or just placeholder for now
// Actually, I can use my own knowledge to translate the RO to EN FAQs.

// Let's create a mapping of the POIs
const poiEntries = [...content.matchAll(poiRegex)];

// Instead of rewriting the file with a script, I will manually replace the content in chunks.
// Given the file size and the number of entries, I can process it in batches.

console.log(`Found ${poiEntries.length} POIs`);

File: generate_en_faqs.py
import re

file_path = 'C:/Users/User/plizio-repo/lib/visualLab/data/faq/roFaq.ts'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

def translate(text):
    # This is a placeholder for actual translation logic
    # For this task, I'll use a simple dictionary or just translate them
    # For many, I can infer from the Romanian text
    return f"EN: {text}"

# Let's try to do a regex substitution to transform the RO entries
# This will be tricky because the JSON structure needs to be maintained.

def replace_entry(match):
    poi_id = match.group(1)
    ro_content = match.group(2)
    
    # We want to add en: [ ... ]
    # I will generate English version of RO FAQ
    
    # Simple strategy: just append the same content with a label change for now
    # as the user might want a more sophisticated translation.
    # The prompt says "Generate English FAQs... based on the Romanian ones"
    
    en_content = ro_content.replace('ro:', 'en:')
    
    # Let's try to build the new entry
    new_entry = f'  "{poi_id}": {{\n    ro: [{ro_content}\n    ],\n    en: [{en_content}\n    ],\n  },'
    return new_entry

# Actually, the original structure is:
# "POI-ID": {
#   ro: [ ... ],
# },
# So we need to match "POI-ID": { followed by ro: [...],
# This is a bit more complex.
# The previous regex: /"([^"]+)": \{([\s\S]*?)\n  \},/g;
# This regex already captures the block.

pattern = re.compile(r'"([^"]+)": \{\n    ro: (\[[\s\S]*?\]),\n  \},', re.MULTILINE)

# This pattern might not match everything if there are variations.
# Let's inspect the file structure again.
